proxy/proxy.go

```diff
@@ -31,41 +31,93 @@ const (
 	proxyBufferSize = 8192
 	proxyErrorFmt   = "proxy: %s"
 
-	// TODO: this should be fine tuned, yet, with benchmarks.
-	// In case it doesn't make a big difference, then a lower value
-	// can be safer, but the default 2 turned out to be too low during
-	// previous benchmarks.
-	//
-	// Also: it should be a parameter.
-	idleConnsPerHost = 64
+	// The default value set for http.Transport.MaxIdleConnsPerHost.
+	DefaultIdleConnsPerHost = 64
+
+	// The default period at which the idle connections are forcibly
+	// closed.
+	DefaultCloseIdleConnsPeriod = 12 * time.Second
 )
 
-type Options uint
+// ProxyFlags control the behavior of the proxy.
+type ProxyFlags uint
 
 const (
-	OptionsNone Options = 0
+	ProxyFlagsNone ProxyFlags = 0
+
+	// ProxyInsecure causes the proxy to ignore the verification of
+	// the TLS certificates of the backend services.
+	ProxyInsecure ProxyFlags = 1 << iota
+
+	// ProxyPreserveOriginal indicates that filters require the
+	// preserved original metadata of the request and the response.
+	ProxyPreserveOriginal
+
+	// ProxyPreserveHost indicates whether the outgoing request to the
+	// backend should use by default the 'Host' header of the incoming
+	// request, or the host part of the backend address, in case filters
+	// don't change it.
+	ProxyPreserveHost
+
+	// ProxyDebug indicates that the current proxy instance will be used as a
+	// debug proxy. Debug proxies don't forward the request to the
+	// route backends, but they execute all filters, and return a
+	// JSON document with the changes the filters make to the request
+	// and with the approximate changes they would make to the
+	// response.
+	ProxyDebug
+)
 
-	// Flag indicating to ignore the verification of the TLS
-	// certificates of the backend services.
-	OptionsInsecure Options = 1 << iota
+// Options are deprecated alias for ProxyFlags.
+type Options ProxyFlags
 
-	// Flag indicating whether filters require the preserved original
-	// metadata of the request and the response.
-	OptionsPreserveOriginal
+const (
+	OptionsNone              = Options(ProxyFlagsNone)
+	OptionsInsecure          = Options(ProxyInsecure)
+	OptionsPreserveOriginal  = Options(ProxyPreserveOriginal)
+	OptionsProxyPreserveHost = Options(ProxyPreserveHost)
+	OptionsDebug             = Options(ProxyDebug)
+)
 
-	// Flag indicating whether the outgoing request to the backend
-	// should use by default the 'Host' header of the incoming request,
-	// or the host part of the backend address, in case filters don't
-	// change it.
-	OptionsProxyPreserveHost
+// Proxy initialization options.
+type ProxyOptions struct {
```

Comment:
Same as above, although I understand the will to keep the old one and hence, naming this one differently.